course6/toyapp-adam/main.cc

```diff
@@ -0,0 +1,59 @@
+#include <displaylib/scene.h>
+#include <displaylib/surface.h>
+#include <displaylib/window.h>
+#include <util/clock.h>
+#include <util/keyboard.h>
+#include <util/mouse.h>
+#include <SDL2/SDL_main.h>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+	Util::Keyboard keyboard;
+	Util::Mouse mouse;
+	Util::Clock clock;
+
+	Display::Window win("Hello", 100, 100, 1280, 720);
+	Display::Surface2D surface2D(640, 360);
+	Display::Surface3D surface3D(surface2D);
+	Display::Scene scene(keyboard, mouse, surface3D);
+
+	SDL_Event event;
+	SDL_SetRelativeMouseMode(SDL_TRUE);
+
+	while(true)
+	{
+		while(SDL_PollEvent(&event))
+		{
+			switch(event.type)
+			{
+				case SDL_KEYDOWN:
+					keyboard.Press(SDL_GetScancodeFromKey(event.key.keysym.sym));
+					break;
+				case SDL_KEYUP:
+					keyboard.Release(SDL_GetScancodeFromKey(event.key.keysym.sym));
+					break;
+				case SDL_MOUSEMOTION:
+					mouse.AddDeltaPosition({event.motion.xrel, event.motion.yrel});
+					break;
+				case SDL_QUIT:
+					goto quit;
+			}
+		}
+
+		if(keyboard.IsPressed(SDL_SCANCODE_END))
+			goto quit;
```

Comment:
```goto``` is not inherently evil. I think your mixing two different quotes attributed to Dijkstra. It was premature optimisation that was evil. In Dijsktra's quotes, gotos where simply the indication of programmers stupidity :)

Use boolean variable to control ```while``` instead.

Reply:
I wasn't trying to quote anyone :) That's quite a strong statement which I wasn't expecting from you. Please don't tell me I'm stupid.

The bool variable would need to be compared to false in every single iteration. Why must I have that? Will the compiler optimize it away? Maybe. But I'm not a fan of premature pessimization either. It took no effort to write it this way.

Actually, what I wanted is to break the SDL event poll loop from within the switch, and figured I don't need stupid additional flags, since I just want to break the hell out of there.